modules/apps/27-interchain-accounts/module.go

```diff
@@ -56,7 +57,12 @@ func (AppModuleBasic) DefaultGenesis(cdc codec.JSONCodec) json.RawMessage {
 
 // ValidateGenesis performs genesis state validation for the IBC interchain acounts module
 func (AppModuleBasic) ValidateGenesis(cdc codec.JSONCodec, config client.TxEncodingConfig, bz json.RawMessage) error {
-	return nil // TODO: https://github.com/cosmos/ibc-go/issues/535
+	var gs types.GenesisState
+	if err := cdc.UnmarshalJSON(bz, &gs); err != nil {
+		return fmt.Errorf("failed to unmarshal %s genesis state: %w", types.ModuleName, err)
```

Comment:
I've used `fmt.Errorf()` as that's what's done in `core` and `transfer`. Is there an issue with returning an sdkerror type here?

Reply:
It is okay to use `fmt.Errorf` for genesis related functions (ie any function run at genesis before the chain starts). These errors will be reported directly to the terminal without being handled by SDK baseapp. These are usually run directly by the CLI

Reply:
Ah okay, thanks for clarifying that :)